tools/migrate_gul14_to_gul17.bash

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+if [ "$#" -eq 0 ]; then
+    echo "Usage: $0 <directories>"
+    echo
+    echo "This script will update C++ source files and meson.build files"
+    echo "in the given directories to migrate from GUL14 to GUL17."
+    echo "The given directories are searched recursively."
+    echo
+    echo "The migration happens in place, so make sure to have a backup"
+    echo "(or a recent git commit) of your files."
+    echo
+    echo "This is a simple text replacement. Some manual modifications may"
+    echo "be necessary to finish the migration and to make the code"
+    echo "compile again."
+    exit 1
+fi
+
+for dir in "$@"; do
+    pushd "$dir"
+
+    source_files=$( \
+        find . -name "*.h" -o -name "*.hpp" -o -name "*.cc" -o -name "*.cpp" | \
+        sed -e '/^.\/build.*\/.*/d' -e '/^.\/debian.*\/.*/d' -e '/^.\/subprojects.*\/.*/d' | \
+        sort )
+    meson_files=$( \
+        find . -name "meson.build" | \
+        sed -e '/^.\/build.*\/.*/d' -e '/^.\/debian.*\/.*/d' -e '/^.\/subprojects.*\/.*/d' | \
+        sort )
+
+    if [ -f "meson.build" ]; then
+        sed -i -E \
+            -e 's/(dependency\s*\(\s*)'"'libgul14'/\1'gul17'"'/g' \
+            -e 's/\[\s*'"'libgul14'"'\s*,\s*'"'libgul_dep'"'\s*\]/['"'gul17', 'gul17_dep']"'/g' \
```

Comment:
I must admit I do not understand this regex.

```shell
$ printf "%s\n" 's/\[\s*'"'libgul14'"'\s*,\s*'"'libgul_dep'"'\s*\]/['"'gul17', 'gul17_dep']"'/g'
s/\[\s*'libgul14'\s*,\s*'libgul_dep'\s*\]/['gul17', 'gul17_dep']/g
```

Ah I see... quoting is too hard.